### PR TITLE
add warning for old PsychoPy versions

### PR DESCRIPTION
**Pull request by eort**
The current warning category is `UserWarning` as none of the others did really fit. 
I hope this is something like what you had in mind.

### CODE REVIEW

File: openexp/_canvas/psycho.py [hunk 1]
```diff
@@ -45,6 +45,18 @@
 	raise osexception(
 		u'PsychoPy is missing the GratingStim() class. Please update your version of PsychoPy! For installation instructions, please visit http://www.psychopy.org/.')
 
+import psychopy, warnings
+from distutils.version import StrictVersion
+try:
+    PsyPy_vers = StrictVersion(psychopy.__version__)
```
Comment (smathot) [inline openexp/_canvas/psycho.py]: You're using `CamelCase`, whereas OpenSesame style is to use `lowercase_with_underscores_if_necessary`.

File: openexp/_canvas/psycho.py [hunk 2]
```diff
@@ -45,6 +45,18 @@
...
+except ValueError:
+    warnings.warn(u'Invalid version number %s of PsychoPy'%psychopy.__version__)
```
Comment (smathot) [inline openexp/_canvas/psycho.py]: It's not visible here, but you're mixing space- and tab-based indentation. That's a *big* no! Only tab-based indentation in OpenSesame. The recommended Python-style (PEP8) is space-based, so you can use that for new projects too. But, in any case, never mix!

File: openexp/_canvas/psycho.py [hunk 3]
```diff
@@ -45,6 +45,18 @@
...
+else:
+	if PsyPy_vers.version < (1, 87):
```
Comment (smathot) [inline openexp/_canvas/psycho.py]: Looking at the warning message, that should probably be `< (1, 84, 1)`?

File: openexp/_canvas/psycho.py [hunk 4]
```diff
@@ -45,6 +45,18 @@
...
+		warnings.warn(u'Your PsychoPy version is too old. You will not be able\
```
Comment (smathot) [inline openexp/_canvas/psycho.py]: It's a matter of style, but I prefer implicit string concatenation over these forced line breaks. In Python `'x' 'y'` becomes `'xy'`. And this works across lines too:

~~~ .python
		warnings.warn(u'Your PsychoPy version is outdated. To draw complex '
			u'shapes that include holes, concavities and such, please update '
			u'PsychoPy to version 1.84.1 or higher.')
~~~

File: openexp/_canvas/psycho.py (hunk 2 above)
Comment (eort) [inline openexp/_canvas/psycho.py]: check

File: openexp/_canvas/psycho.py (hunk 1 above)
Comment (eort) [inline openexp/_canvas/psycho.py]: check

**Comment by smathot** [comment]
Excellent!

Fyi, I cherry-picked the two relevant commits. You merged your own `cv_clear` branch into this branch, and you see the ugly merge commits that resulted? It's best to leave different patches in different branches, and just send independent pull requests for them. If they happen to conflict, then it's up to me to resolve it (but that wasn't the case here).

**Comment by eort** [comment]
Yes, I saw that. It happened before I realized that I merged them, but I hoped that you can cherrypick the important bits. Once it happened, would I be able to undo it (without too much hassle), or would I need to have a fresh start with a new branch?

**Comment by smathot** [comment]
> 
Yes, I saw that. It happened before I realized that I merged them, but I hoped that you can cherrypick the important bits. Once it happened, would I be able to undo it (without too much hassle), or would I need to have a fresh start with a new branch?

That's a pretty advanced git question (the type that I would generally google for), and I'm not sure. You can undo commits with`git revert`, but that actually creates new undo commits, so it makes the situation worse. Creating a clean branch and cherry-picking from your own dirty branch is probably easiest.

**Comment by eort** [comment]
Aha, sounds reasonable. Thanks.

**Pull request closed without merging**